### PR TITLE
support reports with bzip2,xz compression including gz

### PR DESCRIPTION
**Pull request by kgaikwad**
Fixes - [YUPANA-27](https://projects.engineering.redhat.com/browse/YUPANA-27)

### CODE REVIEW

**Review by ShimShtein — APPROVED** [review]
LGTM

File: yupana/processor/tests_report_processor.py [hunk 1]
```diff
@@ -919,6 +919,35 @@ def test_extract_and_create_slices_success(self):
         }
         self.assertEqual(result, expected_result)
 
+    def test_extract_and_create_slices_success_using_bz2(self):
```
Comment (apuntamb) [inline yupana/processor/tests_report_processor.py:922]: Can we add one more similar test case for `xz` compression too?
Also, might want to specify [line 894](https://github.com/quipucords/yupana/pull/336/files#diff-f2010242e5b432fcf8aac4fc931b3751f5c2ca68066904676e52ea7a1b70dd10R894) with test name containing `using gz` as it's no longer the only method of compression?

**Review by apuntamb — CHANGES_REQUESTED** [review]
Looks great! Will test it now locally. Just added one small inline comment.

**Comment by apuntamb** [comment]
Tested locally, works as expected!

**Review by apuntamb — APPROVED** [review]
Looks ready to get in as test failures are no longer occurring. :+1:

**Comment by kgaikwad** [comment]
@apuntamb,
Weird! This test is irrelevant to the change but started failing against the PR & 
on master also I was getting the same test failure.  Could you confirm the test behaviour against master on your side?

Today while debugging on test failure, test passed on local for me but here it is still failing. 
I would suggest to merge this change & I will look into test failure separately as similar behaviour against master too.

**Comment by apuntamb** [comment]
@kgaikwad I agree, but the tests are a Required Check for merging the PR and it won't allow me to merge until the check is green :(

**Comment by ShimShtein** [comment]
@kgaikwad, sometimes the order of the tests is changing and causing random failures. Another thought: the failing test name implies you are testing an async action, maybe the action takes longer than expected?

**Comment by kgaikwad** [comment]
@apuntamb @ShimShtein,
Test is failing at this line [L359](https://github.com/quipucords/yupana/blob/master/yupana/processor/tests_report_processor.py#L359).

On local somehow tests are green so I am facing difficulty in debugging the test failure. 
My assumption from test:
Initially, self.processor.report_platform_id is `None` after execution it expects that value matches with `self.report_record.report_platform_id`. 
Inside test, it gives call to method `await self.processor.delegate_state()` where it checks where processor's current state exists inside `state_functions` (key -> function mapping).
If it doesn't exist then it gives call to `reset_variables()` [here](https://github.com/quipucords/yupana/blob/0104dd097adc85765e7a570c77070dba50a0d06e/yupana/processor/abstract_processor.py#L133) 
Due to some delay, it fails to get `state` value from `processor` & results into `reset_variables()`. 
By adding small sleep test is passing now.

I understand this is very hacky way to fix the test but not getting how to fix such behavior. 
Please let me know your thoughts on it.

**Review by apuntamb — APPROVED** [review]
I guess for now, as a workaround, adding `sleep` would be ok.

**Comment by patilsuraj767** [comment]
Thank you @kgaikwad :+1: Merging this.

**Pull request merged**